atef/widgets/config/find_replace.py

```diff
@@ -0,0 +1,1009 @@
+            return
+
+        edit_row_widget: TemplateEditRowWidget = self.edits_table.cellWidget(
+            selected_ranges[0].topRow(), 0
+        )
+
+        if not isinstance(edit_row_widget, TemplateEditRowWidget):
+            # placeholder text if nothing is selected
+            l_item = QtWidgets.QListWidgetItem("select an edit or click an edit's"
+                                               "refresh button to show details.")
+            return
+        elif not edit_row_widget.get_details_rows():
+            l_item = QtWidgets.QListWidgetItem(
+                'provide search and replace text to show details...'
+            )
+            self.details_list.addItem(l_item)
+            return
+
+        row_widget: FindReplaceRow
+        for row_widget in edit_row_widget.get_details_rows():
+            l_item = QtWidgets.QListWidgetItem()
+            l_item.setSizeHint(QtCore.QSize(row_widget.width(), row_widget.height()))
+            self.details_list.addItem(l_item)
+            self.details_list.setItemWidget(l_item, row_widget)
+
+            row_widget.remove_item.connect(partial(self.remove_item_from_details, l_item))
+
+    def remove_item_from_details(self, item: QtWidgets.QListWidgetItem) -> None:
+        """ remove an item from the details list """
+        self.details_list.takeItem(self.details_list.row(item))
+
+
+class TemplateEditRowWidget(DesignerDisplay, QtWidgets.QWidget):
+    """
+    A widget for specifying the information for find/replace actions
+
+    Each edit can correspond to multiple ``FindReplaceAction``'s
+    Generates ``FindReplaceRow``'s for each ``FindReplaceAction``
+    """
+    button_box: QtWidgets.QDialogButtonBox
+    child_button: QtWidgets.QPushButton
+
+    setting_button: QtWidgets.QToolButton
+    regex_button: QtWidgets.QToolButton
+    case_button: QtWidgets.QToolButton
+
+    search_edit: QtWidgets.QLineEdit
+    replace_edit: QtWidgets.QLineEdit
+
+    filename = 'template_edit_row_widget.ui'
+
+    def __init__(
+        self,
+        *args,
+        data=None,
+        orig_file: Union[ConfigurationFile, ProcedureFile],
+        **kwargs
+    ) -> None:
+        # Expected SimpleRowWidgets are DataWidgets, expecting a dataclass
+        super().__init__(*args, **kwargs)
+        self.orig_file = orig_file
+        self.match_paths: Iterable[List[Any]] = []
+        self.actions: List[FindReplaceAction] = []
+        self._match_fn = lambda x: False
+        self._replace_fn = lambda x: x
+        self.setup_ui()
+
+    def setup_ui(self):
+        self.child_button.hide()
+        refresh_button = self.button_box.button(QtWidgets.QDialogButtonBox.Retry)
+        refresh_button.clicked.connect(self.refresh_paths)
+        refresh_button.setText('')
+        refresh_button.setToolTip('refresh edit details')
+        refresh_button.setIcon(qta.icon('ei.refresh'))
+
+        apply_button = self.button_box.button(QtWidgets.QDialogButtonBox.Apply)
```

Comment:
If I click 'apply all' after having changed a device in the template, I'd have expected to see the device/signal list above change:
<img width="532" alt="image" src="https://github.com/pcdshub/atef/assets/5139267/6b5a76c6-8e5a-4a54-9e13-9211f638994a">